tests/integration/test_factory.py

```diff
@@ -43,7 +37,9 @@ def setUp(self):
             self.gitlab_service,
         ]
 
+    @record_requests()
     def test_get_project_github(self):
+        self.init_services()
```

Comment:
Can we move this to the `SetUp` method so we don't need to put this into all test methods?

Reply:
> Can we move this to the `SetUp` method so we don't need to put this into all test methods?

I wanted that as well, but found it is impossible, because decorators are just around test functions, and setUp is done before tests executions. so services initialization is out, and when running without tokens it led to missing authorizations for setup. I wanted to decorate whole class, but it is impossible in case unittests, because tests are then not found  from some reason. maybe something connected that unittest classes are not intantiated, or something similar. 

This part is little bit tricky and previous solution, was more elegant for this, because it changed modules before anything in tests. But it makes sense that it works how it works :-)

Do you any suggestion how to deal with that? I cannot also decorate setup, because it changes cassette, so it will not be then part of cassette of test.